core/store/src/trie/mod.rs

```diff
@@ -319,20 +366,28 @@ impl TrieCachingStorage {
         // TODO defend from huge values in cache
         TrieCachingStorage { store, cache: Arc::new(Mutex::new(SizedCache::with_size(10000))) }
     }
+}
 
+impl TrieStorage for TrieCachingStorage {
     fn retrieve_raw_bytes(&self, hash: &CryptoHash) -> Option<(Vec<u8>)> {
-        let mut guard = self.cache.lock().expect(POISONED_LOCK_ERR);
-        if let Some(val) = (*guard).cache_get(hash) {
-            val.clone()
-        } else {
-            let result = if let Ok(Some(bytes)) = self.store.get(COL_STATE, hash.as_ref()) {
-                Some(bytes)
+        let result = {
+            let mut guard = self.cache.lock().expect(POISONED_LOCK_ERR);
```

Comment:
I think for poisoned locks it makes sense to just unwrap. If lock is poisoned the program will fail with verbose message anyway.

Reply:
We have this pattern everywhere, it's for readability I think

Reply:
I asked to introduce this pattern before. However, now I realized the following:
* If the `Error` that we are unwrapping implements `Debug` the result of calling unwrap on it will be panic with that error message printed.
* If we say `expect("some custom error message")` then the result is panic with the custom error message printed.

So unless there is any additional information to say, besides the fact that "lock is poisoned" there is no real reason to use `expect` since unwrap will have the same effect.

Reply:
It's a code readability pattern - expect() with an explanation that it can never fail or that it's impossible to recover from the error. More explicit than unwrap()

Reply:
`unwrap()` implies exactly the same semantics. Anywhere in the code if you see `unwrap` it implies that this should never fail. Are you saying the word `unwrap` is less descriptive than the word `expect`?

Reply:
``` let mut guard = self.cache.lock().expect(POISONED_LOCK_ERR); ```
or
``` let mut guard = self.cache.lock().unwrap(); // the only possible failure is poisoned lock```